test/coverage/datamodel/UUriSerializerTest.cpp

```diff
@@ -23,16 +23,167 @@ class TestFixture : public testing::Test {
+	std::string actualUUri =
+	    uprotocol::datamodel::serializer::uri::AsString::serialize(*testUUri);
+	ASSERT_EQ(expectedUUri, actualUUri);
+}
+
+// TODO: Enable this test once validator for UUri is implemented.
+// TEST_F(TestUUriSerializaerFixture, SerializeEmptyUUriToString) {
+// 	uprotocol::v1::UUri testUUri ;
+// 	std::string expectedUUri = "";
+// 	std::string actualUUri 	 =
+// uprotocol::datamodel::serializer::uri::AsString::serialize(testUUri);
+// 	ASSERT_EQ(expectedUUri, actualUUri);
+// }
+
+TEST_F(TestUUriSerializaerFixture, DeSerializeUUriStringWithScheme) {
+	std::string uuriAsString = "up://192.168.1.10/1100/FF/8000";
+	std::string expectedAuthority = "192.168.1.10";
+	uint32_t expectedUEID = 0x1100;
+	uint32_t expectedMajorVersion = 0xFF;
+	uint32_t expectedResourceID = 0x8000;
+
+	uprotocol::v1::UUri uri =
+	    uprotocol::datamodel::serializer::uri::AsString::deserialize(
+	        uuriAsString);
+	ASSERT_EQ(expectedAuthority, uri.authority_name());
+	ASSERT_EQ(expectedUEID, uri.ue_id());
+	ASSERT_EQ(expectedMajorVersion, uri.ue_version_major());
+	ASSERT_EQ(expectedResourceID, uri.resource_id());
+}
+
+TEST_F(TestUUriSerializaerFixture, DeSerializeUUriStringWithoutScheme) {
+	std::string uuriAsString = "//192.168.1.10/1100/FF/8000";
+	std::string expectedAuthority = "192.168.1.10";
+	uint32_t expectedUEID = 0x1100;
+	uint32_t expectedMajorVersion = 0xFF;
+	uint32_t expectedResourceID = 0x8000;
+
+	uprotocol::v1::UUri uri =
+	    uprotocol::datamodel::serializer::uri::AsString::deserialize(
+	        uuriAsString);
+	ASSERT_EQ(expectedAuthority, uri.authority_name());
+	ASSERT_EQ(expectedUEID, uri.ue_id());
+	ASSERT_EQ(expectedMajorVersion, uri.ue_version_major());
+	ASSERT_EQ(expectedResourceID, uri.resource_id());
+}
+
+TEST_F(TestUUriSerializaerFixture, DeSerializeEmptyUUriString) {
+	std::string uuriAsString = "";
+	std::string expectedAuthority = "";
+	uint32_t expectedUEID = 0x0;
+	uint32_t expectedMajorVersion = 0x0;
+	uint32_t expectedResourceID = 0x0;
+
+	uprotocol::v1::UUri uri =
+	    uprotocol::datamodel::serializer::uri::AsString::deserialize(
+	        uuriAsString);
+	ASSERT_EQ(expectedAuthority, uri.authority_name());
+	ASSERT_EQ(expectedUEID, uri.ue_id());
+	ASSERT_EQ(expectedMajorVersion, uri.ue_version_major());
+	ASSERT_EQ(expectedResourceID, uri.resource_id());
+}
+
+TEST_F(TestUUriSerializaerFixture, DeSerializeUUriStringWithNoAuthority) {
+	std::string uuriAsString = "/1100/FF/8000";
+	std::string expectedAuthority = "";
+	uint32_t expectedUEID = 0x1100;
+	uint32_t expectedMajorVersion = 0xFF;
+	uint32_t expectedResourceID = 0x8000;
+
+	uprotocol::v1::UUri uri =
+	    uprotocol::datamodel::serializer::uri::AsString::deserialize(
+	        uuriAsString);
+	ASSERT_EQ(expectedAuthority, uri.authority_name());
+	ASSERT_EQ(expectedUEID, uri.ue_id());
+	ASSERT_EQ(expectedMajorVersion, uri.ue_version_major());
+	ASSERT_EQ(expectedResourceID, uri.resource_id());
+}
+
+TEST_F(TestUUriSerializaerFixture,
```

Comment:
I have observed this weird formatting after running ./lint/clang-format.sh
This has happened because I had single line comment after the expression eg: 

uuriAsString =  "//192.168.1.10/FF/8000";  // UE ID is missing. Provided 3 arguments instead of 4 when authority exist

is converted to :

uuriAsString =
	    "//192.168.1.10/FF/8000";  // UE ID is missing. Provided 3 arguments
	                               // instead of 4 when authority exist.

Reply:
I strongly recommend not using comments at the end of a line of code. Put them on the line above the code being documented.

Reply:
Done